src/App.js

```diff
@@ -0,0 +1,149 @@
+          </Navbar.Collapse>
+        </Container>
+      </Navbar>
+
+      <div className="App">
+        <header className="preview">
+          <img src={logo} className="App-logo" id="img-preview" alt="logo" />
+          <p>Drag and Drop or Click to Upload Images</p>
+          <FileUploader
+            name="file"
+            accept={fileTypes}
+            types={fileTypes}
+            multiple={false}
+            classes="Drop-zone"
+            handleChange={(file) => {
+              setImage(file);
+              const reader = new FileReader();
+              reader.onload = (e) => {
+                document.getElementById("img-preview").src = e.target.result;
+              };
+              reader.readAsDataURL(file);
+              document.getElementById("img-preview").style.display = "block";
+              document.getElementById("img-preview").title = file.name;
+            }}
+            label={"Upload or drop a file right here"}
+          />
+          <OverlayTrigger
+            overlay={
+              <Tooltip id="tooltip-disabled">
+                {!image
+                  ? "Select Image to Upload"
+                  : !(token && repository)
+                  ? "Add Token and Repository"
+                  : "Click to Upload"}
+              </Tooltip>
+            }
+            placement="bottom"
+          >
+            <span className="d-inline-block">
+              <Button
+                variant="success"
+                className="mt-3"
+                disabled={!(token && image && repository)}
+              >
+                Upload
+              </Button>
+            </span>
+          </OverlayTrigger>
+        </header>
+        <Modal
+          show={show}
+          onHide={() => {
+            setShow(false);
+          }}
+        >
+          <Modal.Header closeButton>
+            <Modal.Title>GitHub Login</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Form>
+              <Form.Group className="mb-3">
+                <Form.Label>GitHub Token</Form.Label>
+                <Form.Control type="text" id="token" autoFocus />
+              </Form.Group>
+              <Form.Group className="mb-3">
+                <Form.Label>Repository (username/repository_name)</Form.Label>
+                <Form.Control type="text" id="repo" />
+              </Form.Group>
+            </Form>
+          </Modal.Body>
+          <Modal.Footer>
+            <Button
+              variant="secondary"
+              onClick={() => {
+                setShow(false);
+              }}
+            >
+              Close
+            </Button>
+            <Button
+              variant="success"
+              onClick={() => {
+                if (document.getElementById("token").value === "") {
+                  document.getElementById("token").focus();
+                  return;
+                }
+                if (document.getElementById("repo").value === "") {
```

Comment:
Here, you can use more validation with regex like;
```
 [A-za-z0-9_-]+\/[A-za-z0-9_-]+
```

Reply:
Anything now that is not following the `(username/repository_name)` and `(username/repositoryName)` will not be allowed to submit the form. 

Lastly, I have added the following initializations to make the statements easier to read.

```js
let token = document.getElementById("token");
let repo = document.getElementById("repo");
let validRepo = "[A-za-z0-9_-]+/[A-za-z0-9_-]+";
```